Lib/asyncio/sslproto.py

```diff
@@ -636,11 +633,12 @@ def _process_write_backlog(self):
                         self._on_handshake_complete)
                     offset = 1
                 else:
-                    ssldata = self._sslpipe.shutdown(self._finalize)
+                    ssldata = self._sslpipe.shutdown()
+                    self._feed_ssl_data(ssldata)
+                    self._finalize()
                     offset = 1
```

Comment:
There is no need.  When we return - this code is Omitted :
`del self._write_backlog[0]` 
If `_process_write_backlog` is called after shutdown, we ll do shutdown again. If we want to return here we need to add `del` either. I think this may complicate this method even more.

Maybe something like that [gist](https://gist.github.com/c985ea676eb8f920c7dfa1c517a688bb), but code with write  blocked on a read is not tested, i will suggest to not make more changes.

I am planing to add more unittests